### PR TITLE
[Option 1] Add TerminationHook and FloRunner.Builder

### PR DESCRIPTION
**Pull request by NatashaL**

### CODE REVIEW

File: flo-runner/src/main/resources/reference.conf [hunk 1]
```diff
@@ -11,3 +11,5 @@ styx.workflow.parameter = "UNKNOWN_PARAMETER"
 styx.workflow.parameter = ${?STYX_PARAMETER}
 styx.workflow.execution = "UNKNOWN_EXECUTION"
 styx.workflow.execution = ${?STYX_EXECUTION_ID}
```
Comment (honnix) [inline flo-runner/src/main/resources/reference.conf]: Heh, didn't recall we have this already. So we might just make our life easier by removing the internal code and reading environment variables.

File: flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java [hunk 2]
```diff
@@ -57,16 +57,56 @@
...
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public static FloRunner createDefault() {
+    return newBuilder().build();
+  }
 
   private FloRunner(Config config) {
```
Comment (honnix) [inline flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java]: We have the opportunity to break backward compatibility, so let's break it badly.

File: flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java [hunk 3]
```diff
@@ -57,16 +57,56 @@
 /**
  * This class implements a top-level runner for {@link Task}s.
  */
-public final class FloRunner<T> {
+public class FloRunner<T> {
 
   private static final Logger LOG = LoggerFactory.getLogger(FloRunner.class);
+  private static final TerminationHookFactory DEFAULT_TERMINATION_HOOK_FACTORY = () ->
```
Comment (honnix) [inline flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java]: It doesn't feel we need a factory. A default hook does nothing but exists should be sufficient.

File: flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java [hunk 4]
```diff
@@ -81,6 +121,10 @@ private static Config defaultConfig() {
    * @return a {@link Result} with the value and throwable (if thrown)
    */
   public static <T> Result<T> runTask(Task<T> task, Config config) {
+    if (config.getBoolean("termination.logging")) {
```
Comment (honnix) [inline flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java]: We could make this method non-static. Would that be easier for the internal part?

**Comment by NatashaL** [comment]
Opted for https://github.com/spotify/flo/pull/69 instead

**Pull request closed without merging**